### PR TITLE
Verify catalog schema fieldNames is a subset of DB schema

### PR DESCRIPTION
**Pull request by nguyenaiden**
## What
[Issue #23428](https://app.zenhub.com/workspaces/db--dw-source-connectors-6333360e0a41155061efbcbd/issues/gh/airbytehq/airbyte/23428)
## How
Catalog <> end DB schema comparison now verify if Catalog column keys are a subset of the end DB schema keys. Moved away from full object comparison.
*Note*:  Will potentially update this method at some point to improve testability for unit tests of the `DbSourceDiscoverUtil` class
## 🚨 User Impact 🚨
Sync on CDC mode for DB sources will no longer get incorrect `LOGGER` warnings (see below) for schema change.
`2023-02-23 12:58:36 source > WARN i.a.i.s.r.AbstractDbSource(logSourceSchemaChange):208 Source schema changed for table  public.blah! Actual schema: {"type":"object","properties":{"updated_at":{"type":"string","format":"date-time","airbyte_type":"timestamp_without_timezone"},"id":{"type":"number","airbyte_type":"integer"}}}. Catalog schema:  {"type":"object","properties":{"id":{"type":"number","airbyte_type":"integer"},"updated_at":{"type":"string","airbyte_type":"timestamp_without_timezone","format":"date-time"},"_ab_cdc_lsn":{"type":"number"},"_ab_cdc_deleted_at":{"type":"string"},"_ab_cdc_updated_at":{"type":"string"}}}
`

### CODE REVIEW

File: airbyte-ci/connectors_ci/metadata_service/lib/metadata_service/models/src/AllowedHosts.yaml
Comment (prateekmukhedkar) [inline airbyte-ci/connectors_ci/metadata_service/lib/metadata_service/models/src/AllowedHosts.yaml:13]: why this change?

File: airbyte-ci/connectors_ci/metadata_service/lib/tests/fixtures/invalid/metadata_catalog_no_id_override.yaml
Comment (prateekmukhedkar) [inline airbyte-ci/connectors_ci/metadata_service/lib/tests/fixtures/invalid/metadata_catalog_no_id_override.yaml:21]: Looks like all the files in this folder are showing with whitespace changes. Are they really needed?

File: airbyte-ci/connectors_ci/metadata_service/lib/metadata_service/models/src/AllowedHosts.yaml
Comment (nguyenaiden) [reply airbyte-ci/connectors_ci/metadata_service/lib/metadata_service/models/src/AllowedHosts.yaml:13]: This was an automated change. I'm not sure why it came up.

File: airbyte-ci/connectors_ci/metadata_service/lib/tests/fixtures/invalid/metadata_catalog_no_id_override.yaml
Comment (nguyenaiden) [reply airbyte-ci/connectors_ci/metadata_service/lib/tests/fixtures/invalid/metadata_catalog_no_id_override.yaml:21]: Same as above. I only made changes in 1 file, and the rest were automatically committed. Commit: https://github.com/airbytehq/airbyte/pull/24207/commits/b2e4570f3f411b787e9e98b590377c79b36aad32

File: airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/DbSourceDiscoverUtil.java
Comment (prateekmukhedkar) [inline airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/DbSourceDiscoverUtil.java:60]: isn't `catalogSchema.get(fieldName)` the same as `fieldName` or did you mean to compare with the latter here?

**Comment by prateekmukhedkar** [comment]
@nguyenaiden in order for connectors to use this change they will need to be published. (similar pattern how you did it in https://github.com/airbytehq/airbyte/pull/24000). Refer https://github.com/airbytehq/airbyte/pull/24207#issuecomment-1474371232 for affected connectors.

File: airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/DbSourceDiscoverUtil.java
Comment (prateekmukhedkar) [reply airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/DbSourceDiscoverUtil.java:60]: Possible to write a unit test for this?

Comment (nguyenaiden) [reply airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/DbSourceDiscoverUtil.java:60]: I wanted to compare the k:v pair within the `properties` key. e.g:

_Catalog:_
```
{
   "type":"object",
   "properties":{
      "updated_at":{
         "type":"string",
         "airbyte_type":"timestamp_without_timezone",
         "format":"date-time"
      },
      "id":{
         "type":"number",
         "airbyte_type":"integer"
      }
   }
}
```
_currentJson:_
```
{
   "type":"object",
   "properties":{
      "id":{
         "type":"number",
         "airbyte_type":"integer"
      },
      "updated_at":{
         "type":"string",
         "airbyte_type":"timestamp_without_timezone",
         "format":"date-time"
      },
      "_ab_cdc_lsn":{
         "type":"number"
      },
      "_ab_cdc_deleted_at":{
         "type":"string"
      },
      "_ab_cdc_updated_at":{
         "type":"string"
      }
   }
}
```

**Comment by nguyenaiden** [comment]
@prateekmukhedkar, +1 on that. Waiting for PR approval before publishing the three Dbs mentioned in the ticket.

File: airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/DbSourceDiscoverUtil.java
Comment (akashkulk) [inline airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/DbSourceDiscoverUtil.java:53]: I know it's not related to your change - but could you update the comment for `logSourceSchemaChange` to say : "This method logs schema drift between source table and the catalog. This can happen if (i) underlying table schema changed between syncs or (ii) The source connector's mapping of data types to Airbyte types changed between runs".

**Review by akashkulk — APPROVED** [review]

File: airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/DbSourceDiscoverUtil.java
Comment (akashkulk) [inline airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/DbSourceDiscoverUtil.java:69]: Just add a quick comment on what we're doing here (why we're ignoring the CDC metadata fileds)

**Review by akashkulk — APPROVED** [review]

File: airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/DbSourceDiscoverUtil.java
Comment (nguyenaiden) [reply airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/DbSourceDiscoverUtil.java:71]: +1. On it.

**Comment by subodh1810** [comment]
/publish connector=connectors/source-mssql-strict-encrypt run-tests=false

**Pull request merged**
